hudi-platform-service/hudi-metaserver/hudi-metaserver-client/src/main/java/org/apache/hudi/metaserver/client/HoodieMetaserverClientImp.java

```diff
@@ -109,7 +110,7 @@ public void createTable(Table table) {
   @Override
   public List<HoodieInstant> listInstants(String db, String tb, int commitNum) {
     return exceptionWrapper(() -> this.client.listInstants(db, tb, commitNum).stream()
-        .map(EntityConversions::fromTHoodieInstant)
+        .map(EntityConversions::fromTHoodieInstant).sorted(Comparator.comparing(HoodieInstant::getTimestamp))
```

Comment:
Should this be reversely sorted based on the server side logic (`ORDER BY ts DESC`)?

Reply:
@yihua Thank you for your reply!

Yes, you are right. I had changed `24` to `-1` in the constructor of `HoodieMetaserverBasedTimeline`, so the test was successful in my local env.

If there is the `limit clause` and the `limit > 0`, We won't get the latest instants.

So, we need to sort the instants by instant time in descending order based on the server-side logic.

Reply:
I fixed the code to use the reversed order as discussed.